packages/core/src/browser/quick-open/quick-input-service.ts

```diff
@@ -149,9 +150,23 @@ export class QuickInputService {
                     label,
                     run: mode => {
                         if (!error && mode === QuickOpenMode.OPEN) {
-                            this.onDidAcceptEmitter.fire(undefined);
-                            resolve(currentText);
-                            return true;
+                            if (validateInput && lookFor !== undefined) {
+                                Promise.resolve(validateInput(lookFor))
+                                    .then(err => {
```

Comment:
```suggestion
                                Promise.reject(validateInput(lookFor))
                                    .catch(err => {
```

It looks a bit funny to have a `.then` with the argument `err`, but I won't insist on the change.

Reply:
But yes, `errorMessage` sounds more natural, fixed.